### DISCUSSION TITLE
Why dotnet team prefers abstract classes over interfaces?

### DISCUSSION THREAD

**Discussion by dex3r** [dotnet/runtime#84120] [Q&A] [answered]
In the issue #36617 there is a discussion between creating an abstract class `TimeProvider` vs an interface `ITimeProvider`. 

@terrajobst [says this](https://github.com/dotnet/runtime/issues/36617#issuecomment-1489360375):

> If the request is to make this design interface based I'll bluntlys state that this is off the table, for the reasons we've stated multiple times (abstract base classes are much easier to evolve than interfaces).

I cannot find any more resources stating that. Why is "abstract base classes much easier to evolve than interfaces"? Where can I read more about that?

**Comment by stephentoub** [comment, marked as answer]
https://github.com/dotnet/runtime/issues/36617#issuecomment-1488795826

> You can add new virtuals to a base class without concern for the diamond problem that is present for Default Interface Methods (which also aren't supported on older versions of .NET). For the core libraries, that's an issue we take seriously; for more niche interfaces that are less likely to be inherited, we've started to accept some risk there, but as interfaces become more core, we're much less likely to use DIMs.
